TombEngine/Game/control/box.cpp

```diff
@@ -820,45 +813,46 @@ void CreatureDie(short itemNumber, bool explode)
 
 int BadFloor(int x, int y, int z, int boxHeight, int nextHeight, short roomNumber, LOTInfo* LOT)
 {
-	FloorInfo* floor = GetFloor(x, y, z, &roomNumber);
+	auto floor = GetFloor(x, y, z, &roomNumber);
 	if (floor->Box == NO_BOX)
-		return true;
+		return 1;
 
 	if (LOT->IsJumping)
-		return false;
+		return 0;
 
-	if (g_Level.Boxes[floor->Box].flags & LOT->BlockMask)
-		return true;
+	auto box = &g_Level.Boxes[floor->Box];
+	if (box->flags & LOT->BlockMask)
+		return 1;
 
-	int height = g_Level.Boxes[floor->Box].height;
+	int height = box->height;
 	if ((boxHeight - height) > LOT->Step || (boxHeight - height) < LOT->Drop)
-		return true;
+		return 1;
 
 	if (boxHeight - height < -LOT->Step && height > nextHeight)
-		return true;
+		return 1;
 
 	if (LOT->Fly != NO_FLYING && y > (height + LOT->Fly))
-		return true;
+		return 1;
 
-	return false;
+	return 0;
 }
 
 int CreatureCreature(short itemNumber)  
 {
-	auto* item = &g_Level.Items[itemNumber];
-	auto* object = &Objects[item->ObjectNumber];
+	auto item = &g_Level.Items[itemNumber];
```

Comment:
DO NOT remove asterisks after `auto` if it results in pointer. This is a violation of coding conventions!